lib/quicrq_internal.h

```diff
@@ -371,6 +376,64 @@ typedef struct st_quicrq_notify_url_t {
     uint8_t* url;
 } quicrq_notify_url_t;
 
+/* Context representing unidirectional streams*/
+struct st_quicrq_uni_stream_ctx_t {
+    struct st_quicrq_uni_stream_ctx_t* next_stream;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream;
+    struct st_quicrq_cnx_ctx_t* cnx_ctx;
+    /* Source from which data is read and sent on the stream. */
+    quicrq_media_source_ctx_t* media_source;
+    struct st_quicrq_uni_stream_ctx_t* next_stream_for_source;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream_for_source;
+
```

Comment:
I don't think you need these 3 pointers. The media source maps to the control stream, and you already have a pointer to the control stream.

Reply:
May be we should start from fixing  this structure